koku/masu/database/gcp_report_db_accessor.py

```diff
@@ -588,6 +588,18 @@ def delete_ocp_on_gcp_hive_partition_by_day(self, days, gcp_source, ocp_source,
                             raise err
 
     def get_openshift_on_cloud_matched_tags(self, gcp_bill_id, ocp_report_period_id):
+        # do an initial check to see if there are any matching enabled keys between OCP & GCP
+        match_sql = f"""
+            SELECT COUNT(*) FROM {self.schema}.reporting_gcpenabledtagkeys as gcp
+                INNER JOIN {self.schema}.reporting_ocpenabledtagkeys as ocp ON gcp.key = ocp.key;
+        """
+        with connection.cursor() as cursor:
+            cursor.db.set_schema(self.schema)
+            cursor.execute(match_sql)
+            results = cursor.fetchall()
+            if results[0][0] < 1:
+                LOG.info(f"No matching enabled keys for OCP on GCP {self.schema}")
+                return None
```

Comment:
The return statement below returns a list. Can we replace this `None` with an empty list to keep the return types the same?

Reply:
Also, is this the correct method to update? Shouldn't we be updating get_openshift_on_cloud_matched_tags_trino?

Reply:
So, for context on the initial download this function will return []. We check to see if we get an empty result, we then call the trino sql which takes a long time. However, if they have no enabled tag keys that match then it will always return an empty list. I see your point about about `None` being confusing though and switch it over to the string `skip` to bypass the trino sql. Thanks for the great feedback it has been super helpful lately during these long days and much appreciated. 🙇

Reply:
So I'm not really a fan of returning `"skip"` either. Can we move this check functionality to a new method and then use that result to gate calling the `get_openshift_on_cloud_matched_tags` method?

Reply:
Yeah, I think that is an approach we could implement for this let me take a swing at it and ill ping ya when I got something figured out.

Reply:
I think we will have to implement a method like that for all the other providers as well, but we should be able to add it easily. Let me check to make sure the logic of skipping the trino tag matching is consistent with the other providers real quick by checking their SQL.